pre_commit_hooks/check_added_large_files.py

```diff
@@ -1,24 +1,26 @@
 import argparse
-import json
 import math
 import os
 from typing import Optional
 from typing import Sequence
 from typing import Set
 
 from pre_commit_hooks.util import added_files
-from pre_commit_hooks.util import CalledProcessError
 from pre_commit_hooks.util import cmd_output
 
 
-def lfs_files() -> Set[str]:
-    try:
-        # Introduced in git-lfs 2.2.0, first working in 2.2.1
-        lfs_ret = cmd_output('git', 'lfs', 'status', '--json')
-    except CalledProcessError:  # pragma: no cover (with git-lfs)
-        lfs_ret = '{"files":{}}'
+def filter_lfs_files(filenames: Set[str]) -> None:
+    """Remove files tracked by git-lfs from the set."""
+    if not filenames:
+        return
 
-    return set(json.loads(lfs_ret)['files'])
+    check_attr_ret = cmd_output(
+        'git', 'check-attr', 'filter', '-z', '--', *filenames,
```

Comment:
this could exceed commandline lengths -- this should probably use `--stdin`

Reply:
Done. Since `pre_commit_hooks.util.cmd_output` doesn't allow passing an input, I switched to use `subprocess.run()`.